src/index.js

```diff
@@ -110,39 +121,26 @@ function loaderForModel(model, attribute, options = {}) {
   let cacheKey = getCacheKey(model, attribute, options);
 
   if (!cache.has(cacheKey)) {
-    if (options.limit) {
```

Comment:
I've refactored this down to two cases, instead of one, since the !limit and keys.length === 1 code was the same